lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -153,16 +154,18 @@ export class FileAutoComplete extends React.Component {
         });
     }
 
-    updateIfDirectoryChanged(value) {
+    updateIfDirectoryChanged(value, cb) {
         const directory = this.getDirectoryForValue(value);
         const changed = directory !== this.state.directory;
         if (changed && this.state.directoryFiles !== null) {
-            this.setState({
-                displayFiles: [],
-                directoryFiles: null,
-                directory: directory,
-                open: false,
-            });
+            this.setState(() => {
+                return {
```

Comment:
No need for the `{ return; }`

Reply:
In this case it's needed i think, as that's the functional version.

Reply:
Right, but `() => { foo: null }` is still a function. :-)

Reply:
In general, `() => expression` is fully equivalent to `() => { return expression }`